### PR TITLE
Add giphy panel to select a gif

### LINKED ISSUE #14: Add giphy integration
Add giphy to make easier to select a gif for the challenge.
https://giphy.com/

### PR DESCRIPTION
**Pull request by namadnuno**
Result:
imagem

Closes #14

### CODE REVIEW

File: src/components/Giphy/SearchPanel.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import withTranslations from '../../translations/withTranslations';
+import Item from './Item';
+
+const SearchPanel = ({ onChange, selected, value: { lang, t } }) => {
+  const [gifs, setGifs] = useState([]);
+  const [query, setQuery] = useState('');
+
+  const getGifs = () => {
+    fetch(
+      `https://api.giphy.com/v1/gifs/search?api_key=IzddqNSUg0KutV54FYGCg1iETKpVUVXI&q=${query}&limit=8&offset=0&lang=${lang}`
```
[the remark this answers is not in the document]
Comment (namadnuno) [reply src/components/Giphy/SearchPanel.js:14]: Yes you are right i will take the language out of there, about the business logic i will change that as well.

File: src/components/Giphy/SearchPanel.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    setQuery(value);
+  }, 150);
+
+  useEffect(() => {
+    getGifs();
+  }, [query]);
+
+  return (
+    <View>
+      <Text style={[styles.title]}>{t('cMedia')}</Text>
```
[the remark this answers is not in the document]
Comment (namadnuno) [reply src/components/Giphy/SearchPanel.js]: done

File: src/components/Giphy/SearchPanel.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      <TextInput
+        style={styles.textInput}
+        placeholder={t('SearchForIt')}
+        onChangeText={handleChange}
+      />
```
[the remark this answers is not in the document]
Comment (namadnuno) [reply src/components/Giphy/SearchPanel.js]: done

File: src/translations/translations.json [hunk 4]
```diff
@@ -36,7 +37,7 @@
     "cDescriptionPlaceholder": "Uma breve descrição do desafio",
     "cHashtag": "Hashtag do desafio",
     "cHashtagPlaceholder": "#hashtag_do_desafio",
-    "cMedia": "Url da media (imagem ou gif)",
+    "cMedia": "Choose an GIF",
```
Comment (rbgoncalves) [inline src/translations/translations.json:40]: This translation is supposed to be in Portuguese

File: src/components/Giphy/styles.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  selectedImg: {
+    width: 150,
+    aspectRatio: 1,
+    alignSelf: 'center',
+    borderColor: '#0096fe',
+    borderWidth: 5,
+    borderRadius: 20,
+    marginBottom: 10,
+    marginRight: 15,
+  },
```
Comment (rbgoncalves) [inline src/components/Giphy/styles.js:45]: these are common, only the color change, we could reuse it and conditionally only toggle the color

File: src/components/Giphy/useGiphy.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  const setDefaultValue = (value) => {
+    if (query === '') getLazy(endpoint, apiKey, value, setGifs);
+  };
+
+  React.useEffect(() => {
+    getLazy(endpoint, apiKey, query, setGifs);
+  }, [query]);
+
+  return [handleQueryChange, setDefaultValue, gifs, query];
+};
```
Comment (rbgoncalves) [inline src/components/Giphy/useGiphy.js:19]: prefer named imports instead of `React.abc`

File: src/components/Giphy/GiphyClient.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+    .then((response) => response.json())
+    .then((response) => {
+      callback(response.data);
+    });
+};
+
+export const getLazy = debounce(
+  (endpoint, apiKey, q, callback) => getGifs(endpoint, apiKey, q, callback),
+  150
+);
```
Comment (rbgoncalves) [inline src/components/Giphy/GiphyClient.js:14]: these should be dispatched in redux, you still having business logic inside the component

File: src/components/Giphy/GiphyContext.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import React, { createContext } from 'react';
+import useGiphy from './useGiphy';
+
+export const GiphyContext = createContext();
+
+export const GiphyProvider = ({ children }) => {
+  const [handleQueryChange, setDefaultValue, gifs] = useGiphy(
+    '/search',
+    'IzddqNSUg0KutV54FYGCg1iETKpVUVXI'
```
Comment (rbgoncalves) [inline src/components/Giphy/GiphyContext.js:9]: also, the API key shouldn't be there

File: src/components/Giphy/styles.js (hunk 5 above)
Comment (namadnuno) [reply src/components/Giphy/styles.js]: done

**Pull request merged**